### PR TITLE
[UXE-3476] feat: adding modal to copy the created domain

### PR DESCRIPTION
**Pull request by lucasmendes21**
### What is the new behavior introduced by this PR?
adding modal to copy the created domain

### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.
https://jam.dev/c/22fe1bb0-1d69-4930-821b-95da3bfe8a83

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: src/views/Domains/FormFields/FormFieldsCreateDomains.vue [hunk 1]
```diff
@@ -39,7 +38,7 @@
   const { value: edgeApplication, errorMessage: errorEdgeApplication } = useField('edgeApplication')
   const { setValue: setEdgeCertificate } = useField('edgeCertificate')
   const { value: mtlsIsEnabled } = useField('mtlsIsEnabled')
-  const { value: domainName } = useField('domainName')
+  useField('domainName')
```
Comment (vinigfer) [inline src/views/Domains/FormFields/FormFieldsCreateDomains.vue:41]: Why we are no longer assigning this to anything anymore?

File: src/views/Home/HomeView.vue [hunk 2]
```diff
@@ -97,6 +97,10 @@
     name: yup
       .string()
       .required('Full Name is a required field')
+      .test('non-numeric', 'Full Name must contain only alphabetic characters', (value) => {
```
Comment (vinigfer) [inline src/views/Home/HomeView.vue:100]: Should this be a part of this PR? It looks like it's out of scope

File: src/views/Domains/FormFields/FormFieldsCreateDomains.vue (hunk 1 above)
Comment (lucasmendes21) [reply src/views/Domains/FormFields/FormFieldsCreateDomains.vue:41]: Porque não usaremos ele mais no FormFieldsCreateDomains eu só preciso que ela seja carregada para eu pegar o nome domain no handle.

File: src/views/Home/HomeView.vue (hunk 2 above)
Comment (lucasmendes21) [reply src/views/Home/HomeView.vue:100]: Isso foi algo que fiz a parte para subir em outro pr que eu fosse subir.

File: src/views/Domains/Dialog/CopyDomainDialog.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      </div>
+
+      <div class="flex justify-start w-full">
+        <PrimeButton
+          icon="pi pi-clone"
+          outlined
+          type="button"
+          aria-label="Copy"
+          label="Copy"
+          :disabled="!domainValue"
```
Comment (matheus-sunderhus) [inline src/views/Domains/Dialog/CopyDomainDialog.vue:47]: criar computed, hasDomain.

File: src/views/Domains/Dialog/CopyDomainDialog.vue [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  import PrimeInputText from 'primevue/inputtext'
+  import PrimeButton from 'primevue/button'
+
+  defineOptions({ name: 'CopyDomainDialog' })
+
+  const dialogRef = inject('dialogRef')
+
+  const params = dialogRef.value.data
+
+  const domainValue = computed({
```
Comment (matheus-sunderhus) [inline src/views/Domains/Dialog/CopyDomainDialog.vue:76]: short sintax:
const domainValue = computed(()=>{return props.domain})

File: src/views/Domains/FormFields/FormFieldsCreateDomains.vue (hunk 1 above)
Comment (matheus-sunderhus) [reply src/views/Domains/FormFields/FormFieldsCreateDomains.vue:41]: pode remover, pois o componente input field internamente usa esse hook para acessar o valor, não sendo mais necessário deixar ele aqui .

File: src/views/Home/HomeView.vue [hunk 5]
```diff
@@ -97,6 +97,10 @@
...
+        const alphaRegex = /[A-zÀ-ž.'-]+[A-zÀ-ž.'-]+/
```
Comment (william-tome) [inline src/views/Home/HomeView.vue:101]: Faltou o espaço
```suggestion
        const alphaRegex = /[A-zÀ-ž.'-]+ [A-zÀ-ž.'-]+/
```

File: src/views/Home/HomeView.vue [hunk 6]
```diff
@@ -97,6 +97,10 @@
...
+        return alphaRegex.test(value)
+      })
       .test('full-name', 'Full Name must include first and last name', (value) => {
         const fullname = value.trim('').split(' ')
         return fullname.length >= 2
```
Comment (william-tome) [inline src/views/Home/HomeView.vue:106]: adicionando o espaço na regex da regra acima já valida esse cenário
```suggestion
```

File: src/views/Home/HomeView.vue [hunk 7]
```diff
@@ -97,6 +97,10 @@
     name: yup
       .string()
       .required('Full Name is a required field')
+      .test('non-numeric', 'Full Name must contain only alphabetic characters', (value) => {
+        const alphaRegex = /[A-zÀ-ž.'-]+[A-zÀ-ž.'-]+/
+        return alphaRegex.test(value)
```
Comment (william-tome) [inline src/views/Home/HomeView.vue:102]:
```suggestion
      .test('non-numeric', 'Full Name must include first and last name', (value) => {
        const alphaRegex = /[A-zÀ-ž.'-]+ [A-zÀ-ž.'-]+/
        return alphaRegex.test(value)
```

**Review by matheus-sunderhus — APPROVED** [review]
